common/persistence/cassandraHistoryPersistence.go

```diff
@@ -143,11 +150,21 @@ func (h *cassandraHistoryPersistence) AppendHistoryEvents(request *AppendHistory
 func (h *cassandraHistoryPersistence) GetWorkflowExecutionHistory(request *GetWorkflowExecutionHistoryRequest) (
 	*GetWorkflowExecutionHistoryResponse, error) {
 	execution := request.Execution
-	query := h.session.Query(templateGetWorkflowExecutionHistory,
-		request.DomainID,
-		*execution.WorkflowId,
-		*execution.RunId,
-		request.NextEventID)
+	var query *gocql.Query
+	if request.FirstEventID > 0 {
+		query = h.session.Query(templateGetWorkflowExecutionHistoryWithStartEventID,
```

Comment:
i don't think this new query is needed. You can always just pass the firstEventID even if it is <=0 in which case it is just noop.

Reply:
adding the yet another query is for backward compatibility during the deployment.
since we do not really care the about that and we have retry, using only the updated version make sense